### PR TITLE
pubsub: add publish with custom attributes sample

### PR DESCRIPTION
**Pull request by anguillanneuf**

### CODE REVIEW

File: pubsub/topics/main.go [hunk 1]
```diff
@@ -193,6 +193,26 @@ func publishThatScales(client *pubsub.Client, topic string, n int) error {
...
 }
 
+func publishCustomAttributes(client *pubsub.Client, topic string) error {
+	ctx := context.Background()
+	// [START pubsub_publish_custom_attributes]
+	t := client.Topic(topic)
+	result := t.Publish(ctx, &pubsub.Message{
+		// data must be a ByteString
+		Data: []byte("Hello world!"),
+		Attributes: map[string]string{"origin": "golang", "username": "gcp"},
```
Comment (broady) [inline pubsub/topics/main.go]: minor style nit: put each k/v pair on a newline

File: pubsub/topics/main.go [hunk 2]
```diff
@@ -193,6 +193,26 @@ func publishThatScales(client *pubsub.Client, topic string, n int) error {
...
+	})
+	// Block until the result is returned and a server-generated
+	// ID is returned for the published message.
+	id, err := result.Get(ctx)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Published message with custom attributes. Msg ID: %v\n", id)
```
Comment (broady) [inline pubsub/topics/main.go]: consistent message:

`Published message with custom attributes; msg ID: %v\n`

File: pubsub/topics/main.go [hunk 3]
```diff
@@ -193,6 +193,26 @@ func publishThatScales(client *pubsub.Client, topic string, n int) error {
 	// [END pubsub_publish_with_error_handling_that_scales]
 }
 
+func publishCustomAttributes(client *pubsub.Client, topic string) error {
+	ctx := context.Background()
+	// [START pubsub_publish_custom_attributes]
+	t := client.Topic(topic)
+	result := t.Publish(ctx, &pubsub.Message{
+		// data must be a ByteString
```
Comment (tbpg) [inline pubsub/topics/main.go]:
```suggestion
		// Data must be a ByteString.
```

Comment (broady) [reply pubsub/topics/main.go]: in fact, i'd just cut this comment (and the one above where it was copied from.

ByteString doesn't mean anything in Go. (`[]byte` is a very common type)

**Pull request merged**